targets/unity-v2/source/Shared/Public/Pipeline/EventSendingStage.cs

```diff
@@ -0,0 +1,112 @@
+﻿#if NET_4_6
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using PlayFab.EventsModels;
+using PlayFab.Logger;
+using PlayFab.SharedModels;
+
+namespace PlayFab.Pipeline
+{
+    /// <summary>
+    /// The event sending stage.
+    /// </summary>
+    internal class EventSendingStage : PipelineStageBase<TitleEventBatch, PlayFabResult<WriteEventsResponse>>
+    {
+        private OneDSEventsAPI oneDSEventsApi;
+        private ILogger logger;
+
+        public EventSendingStage(ILogger logger)
+        {
+            this.logger = logger;
+            this.oneDSEventsApi = new OneDSEventsAPI();
+        }
+
+        /// <summary>
+        /// This method is called by pipeline for each available input item (a batch of events)
+        /// </summary>
+        /// <param name="batch">The input item (a batch of events).</param>
+        protected override void OnNextInputItem(TitleEventBatch batch)
+        {
+            // Send a batch and wait for result
+            var writeEventsRequest = new WriteEventsRequest
+            {
+                Events = batch.Events.Cast<PlayFabEmitEventRequest>().Select(x => x.Event.EventContents).ToList()
+            };
+
+            if (!oneDSEventsApi.IsOneDSAuthenticated)
+            {
+                var authResult = OneDSEventsAPI.GetTelemetryIngestionConfigAsync(new TelemetryIngestionConfigRequest());
+                try
+                {
+                    authResult.Wait(cts.Token);
+                    var response = authResult.Result.Result;
+                    if (response != null)
+                    {
+                        oneDSEventsApi.SetCredentials("o:" + response.TenantId, response.IngestionKey, response.TelemetryJwtToken, response.TelemetryJwtHeaderKey, response.TelemetryJwtHeaderPrefix);
+                    }
+                    else
+                    {
+                        throw new Exception($"Failed to get OneDS authentication token from PlayFab service");
+                    }
+                }
+                catch (Exception e)
```

Comment:
I'm not a fan of putting large blocks of code in try-catch, especially if it seems that the whole point is to just catch the exception thrown 3 lines up (but maybe that's not the only case?)
It's a very dangerous way to code, and it invites problems.
Like, maintaining this code, I have no idea where all the possible exceptions can come from, or why they might happen.

Reply:
If this is LITERALLY just for the exception 3 lines up, then that's TERRIBLY lazy, and should be fixed.

Reply:
No, it's for any exception including CancellationException that may happen at authResult.Wait(...) above. This try-catch is justified. Marking it as resolved.

---
In reply to: [264020062](https://github.com/PlayFab/SDKGenerator/pull/436#discussion_r264020062) [](ancestors = 264020062)